Sources/performance-worker/WorkerService.swift

```diff
@@ -87,7 +116,37 @@ final class WorkerService: Grpc_Testing_WorkerService.ServiceProtocol, Sendable
   ) async throws
     -> GRPCCore.ServerResponse.Stream<Grpc_Testing_WorkerService.Method.RunServer.Output>
   {
-    throw RPCError(code: .unimplemented, message: "This RPC has not been implemented yet.")
+    return ServerResponse.Stream { writer in
+      try await withThrowingTaskGroup(of: Void.self) { group in
+        for try await message in request.messages {
+          switch message.argtype {
+          case let .some(.setup(serverConfig)):
+            let server = try await self.setupServer(serverConfig)
+            group.addTask { try await server.run() }
+
+          case let .some(.mark(mark)):
+            let response = try await self.makeStatsResponse(reset: mark.reset)
+            try await writer.write(response)
+
+          case .none:
+            ()
+          }
+        }
+
+        try await group.next()
+      }
+
+      let server = self.state.withLockedValue { state in
+        state.getServer()
+      }
+
+      guard let server = server else {
+        return [:]
+      }
+
+      server.stopListening()
+      return [:]
+    }
```

Comment:
Let's simplify this a bit:

```suggestion
      server?.stopListening()
      return [:]
    }
```